src/MyEventEmitter.js

```diff
@@ -1,12 +1,80 @@
 'use strict';
 
 export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = new Map()
+  }
+  on(event, callback) {
+    if (this.events.get(event)) {
+      const thisEvent = this.events.get(event)
+      this.events.delete(event)
+      this.events.set(event, [...thisEvent, callback])
+    } else {
+      this.events.set(event, [callback])
+    }
+  }
+  once(event, callback) {
+    const callbackWrapper = () => {
+      callback()
+      this.off(event, callbackWrapper)
+    }
+    if (this.events.get(event)) {
+      const thisEvent = this.events.get(event)
+      this.events.delete(event)
+      this.events.set(event, [...thisEvent, callbackWrapper])
+    } else {
+      this.events.set(event, [callbackWrapper])
+    }
+    return this
+  }
+  off(event, callback) {
+    const eventList = this.events
+      .get(event)
+      .filter((c) => c + "" !== callback + "")
```

Comment:
Why do you convert and compare functions as strings?
Remind me how we store objects in memory? Why do we call real event.off() method with the same function (not copy or else)?